### ISSUE TITLE
ChangeVisibilityTimeoutBatch with VisibilityTimeout of 0 throws error

### ISSUE THREAD

**Issue by atljoseph** [aws/aws-sdk#527]
### Describe the bug

ChangeVisibilityTimeoutBatch does not act in the same way ChangeVisibilityTimeout does.
When 0 is introduced as VisibilityTimeout (a valid value for placing a message back onto the queue), this error is thrown:
```
The request must contain the parameter ChangeMessageVisibilityBatchRequestEntry.VisibilityTimeout
```

### Expected Behavior

ChangeVisibilityTimeoutBatch should accept the same VisibilityTimeout value as ChangeVisibilityTimeout.

### Current Behavior

Error is thrown, when valid value is provided (see description).
```
The request must contain the parameter ChangeMessageVisibilityBatchRequestEntry.VisibilityTimeout
```

### Reproduction Steps

Just send an api call with this input for 
```
&sqs.ChangeMessageVisibilityBatchInput{
	Entries: []types.ChangeMessageVisibilityBatchRequestEntry{
                {
                        Id: "message-id-1",
                        ReceiptHandle: "message-handle-1",
                        VisibilityTimeout: int32(0),
                },
                {
                        Id: "message-id-2",
                        ReceiptHandle: "message-handle-2",
                        VisibilityTimeout: int32(0),
                },
                ...
        },
	QueueUrl: "http://aws.com/random-queue-url",
}
```

Error: 
```
The request must contain the parameter ChangeMessageVisibilityBatchRequestEntry.VisibilityTimeout
```

### Possible Solution

Accept same values in ChangeMessageVisibilityBatch as you do in ChangeMessageVisibility.

### AWS Go SDK V2 Module Versions Used

github.com/aws/aws-sdk-go-v2 v1.17.5
github.com/aws/aws-sdk-go-v2/config v1.18.15
github.com/aws/aws-sdk-go-v2/service/sqs v1.19.17

### Compiler and Version used

go1.19

### Operating System and version

darwin/amd64

**Comment by RanVaknin** [comment]
Hi @atljoseph ,

We have tried getting the SQS team to solve this but they are unable to prioritize this fix.
If you willing, please cut an additional internal ticket to the SQS team via the AWS Console and reference the tracking ticket ID we already have with them `P97774845`, having a customer directly asking for remediation will help prioritization.

Since this is a duplicate of https://github.com/aws/aws-sdk/issues/261 I'll be closing this issue in favor of tracking it there. 

Thanks,
Ran~

**Comment by jfrconley** [comment]
This issue is *still* present

I'm a little confused what has been blocking what seems like a very simple fix, would it be possible to check with SQS team again? @RanVaknin

**Comment by lucix-aws** [comment]
We've had difficulty in the past when communicating with service teams about these issues. Often, from their perspective, the issue doesn't manifest in the SDKs they're familiar with (generally Java, JS, etc.) and so they're hesitant to commit modeling changes because the impact isn't necessarily fully understood. Between that, and other things getting prioritized, these discussions with service teams often start and then eventually stale out.

This particular issue and many like it are only going to affect SDKs with explicit nullability/optionality in their type systems, so generally
* Go v2
* Rust
* Kotlin

We're reframing our strategy when dealing with these service teams going forward, **but for the Go SDK in particular, we're going to just start backfilling modeling fixes locally. So, for this issue, we're going to mark ChangeMessageVisibilityBatchRequestEntry.VisibilityTimeout as `@required` in our own SDK, such that the SDK always sends it.**

That's ultimately a stopgap, but it's minimal effort and it unblocks our customers. We _still_ need to do our best internally and push teams to fix issues like this at the service level, but we shouldn't let perfect be the enemy of good here.

**Comment by lucix-aws** [comment]
@atljoseph @jfrconley https://github.com/aws/aws-sdk-go-v2/pull/2767
